logstreams/src/test/java/io/camunda/zeebe/logstreams/impl/log/SequencerTest.java

```diff
@@ -0,0 +1,340 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.logstreams.impl.log;
+
+import io.camunda.zeebe.logstreams.log.LogAppendEntry;
+import io.camunda.zeebe.scheduler.ActorCondition;
+import io.camunda.zeebe.util.buffer.BufferWriter;
+import java.time.Duration;
+import java.util.List;
+import java.util.concurrent.locks.LockSupport;
+import java.util.stream.IntStream;
+import org.agrona.MutableDirectBuffer;
+import org.assertj.core.api.Assertions;
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.parallel.Execution;
+import org.junit.jupiter.api.parallel.ExecutionMode;
+import org.mockito.Mockito;
+
+@SuppressWarnings("resource")
+@Execution(ExecutionMode.CONCURRENT)
+final class SequencerTest {
```

Comment:
I didn't implement this as parameterized tests because that made them a bit difficult to follow IMO. Instead I added a companion test that writes multiple entries at once everywhere where it made sense. Hope that's okay too.